### PR TITLE
LibUnicode+LibJS+LibWeb: Remove some small OOM propagations

### PR DESCRIPTION
**Pull request by trflynn89**
Work towards #20449.

### CODE REVIEW

File: Userland/Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp [hunk 1]
```diff
@@ -83,6 +83,7 @@ void ReadLoopReadRequest::on_chunk(JS::Value chunk)
         }
 
         m_failure_steps(exception.value());
+        return;
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp]: This seems like a reasonable change but probably belongs in its own PR :)

Comment (trflynn89) [reply Userland/Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp]: Sure - I just added it here because it becomes immediately obvious in the next commit, which shrinks this `if` block down to a single line

Comment (awesomekling) [reply Userland/Libraries/LibWeb/Streams/ReadableStreamDefaultReader.cpp]: Oh hm, fair enough!
